File: auth-service/src/main/java/com/masarnovsky/popugjira/auth/repository/AccountRepository.java
package com.masarnovsky.popugjira.auth.repository;

import com.masarnovsky.popugjira.auth.model.Account;
import org.bson.types.ObjectId;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface AccountRepository extends MongoRepository<Account, ObjectId> {
    Account findByUsername(String username);
    Account findByPublicId(String publicId);
}

File: accounting-service/src/main/kotlin/com/masarnovsky/popugjira/accounting/repository/AccountRepository.kt
package com.masarnovsky.popugjira.accounting.repository

import com.masarnovsky.popugjira.accounting.model.Account
import org.springframework.data.mongodb.repository.MongoRepository

interface AccountRepository : MongoRepository<Account, String> {
    override fun findAll(): List<Account>

    fun findByPublicId(publicId: String): Account
}